portfolio/src/main/java/com/google/sps/servlets/BasketballCourtsServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.BasketballCourtsDatastore;
+
+/** Returns the accessible basketball courts in New York City */
+@WebServlet("/basketball-courts")
+public final class BasketballCourtsServlet extends HttpServlet {
+  
+  private static final Logger log = Logger.getLogger(BasketballCourtsServlet.class.getName());
+  public static final String JSON_FILE_PATH = "/WEB-INF/basketball-courts-nyc.json";
+  List<JsonObject> accessibleCourts;
+
+  @Override
+  public void init() {
+    try(Reader reader = new InputStreamReader(getServletContext().getResourceAsStream(JSON_FILE_PATH))) {
```

Comment:
this is one way of doing it. in that case your "BasketballCourtsDataStore" is no longer a datastore just a parser of JSON courts to BasketballCourts object.

The other way is to pass the ServletContext to your getAccessibleBasketballCourts()  OR make these methods non-static and initialize the `BasketballCourtsDatastore(ServletContext context)` then use getAccessibleCourts(): etc. and the constructor can load the file or something similar.